SingularityBase/src/main/java/com/hubspot/singularity/api/SingularityS3SearchRequest.java

```diff
@@ -0,0 +1,77 @@
+package com.hubspot.singularity.api;
+
+import java.util.Collections;
+import java.util.Set;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.base.Objects;
+import com.google.common.base.Optional;
+import com.wordnik.swagger.annotations.ApiModelProperty;
+
+public class SingularityS3SearchRequest {
+  private final Optional<Long> start;
+  private final Optional<Long> end;
+  private final boolean excludeMetadata;
+  private final boolean listOnly;
+  private final Optional<Integer> maxPerPage;
+  private final Set<ContinuationToken> continuationTokens;
+
+  @JsonCreator
+
+  public SingularityS3SearchRequest(@JsonProperty("start") Optional<Long> start,
+                                    @JsonProperty("end") Optional<Long> end,
+                                    @JsonProperty("excludeMetadata") Optional<Boolean> excludeMetadata,
+                                    @JsonProperty("listOnly") Optional<Boolean> listOnly,
+                                    @JsonProperty("maxPerPage") Optional<Integer> maxPerPage,
+                                    @JsonProperty("continuationTokens") Set<ContinuationToken> continuationTokens) {
+    this.start = start;
+    this.end = end;
+    this.excludeMetadata = excludeMetadata.or(false);
+    this.listOnly = listOnly.or(false);
+    this.maxPerPage = maxPerPage;
+    this.continuationTokens = Objects.firstNonNull(continuationTokens, Collections.<ContinuationToken>emptySet());
```

Comment:
I like the idea of the continuation tokens but having to look up in the set seems a little janky to me -- could we map bucket + prefix to the token as a `Map<String, ContinuationToken>` ?